qutebrowser/browser/commands.py

```diff
@@ -1847,13 +1663,23 @@ def debug_webaction(self, action, count=1):
             action: The action to execute, e.g. MoveToNextChar.
             count: How many times to repeat the action.
         """
-        member = getattr(QWebPage, action, None)
-        if not isinstance(member, QWebPage.WebAction):
+        tab = self._current_widget()
+
+        if tab.backend == tabmod.Backend.QtWebKit:
```

Comment:
The way the function is put in the abstraction API suggests that it works the same way with every backend (like every other function), though that is clearly not the case. If you want to call the function correctly (i.e. working with both backends), you need to check the backend anyway at call site to get the right action, at which point `run_webaction` becomes useless since you already know the backend and you could do (pseudo-code-ish)

```
if backend == QtWebKit:
    action = QWebPage.specificAction
    tab.widget.triggerAction(action)
else:
    action = QWebEnginePage.specificAction
    tab.widget.triggerAction(action)
```

(just noticed that even the original method names are equal, so `run_webaction` is even more useless :wink:).

I see that you're only using it once as part of a debug method, so you might not put that much abstraction into it, but since it may become part of the plugin API (or just used in other places in qutebrowsers code), I don't like this abstraction. Even if you don't have the action as a string, you'd still need the `if`-check to get the right action. It would be better to either remove it from the common API (and just call `triggerAction` directly in the `debug-webaction` method) or rework it to be backend-agnostic (e.g. by using an own enum (or even string parameter) (with Pythonmagic you could even generate the enum automatically)). At least you should put some kind of red flag to the function, e.g. a `debug_` prefix or a warning in the docstring (something like `"This function expects backend-specific actions. Usage of this function in general purpose code is discouraged, you should use the backend-agnostic APIs whenever possible."`)